src/priya/SumAllDigitGivenString.java

```diff
@@ -0,0 +1,31 @@
+//program 2 : sum of all numbers in a give string 
+//input: te123ch9kj8/////output : 123 + 9 + 8 = 140
+
+package priya;
+
+public class SumAllDigitGivenString {
+
+	void calculateDigitGivenString(String str) {
+		String num ="";
+		int sum =0;
+		for(int index =0; index< str.length(); index++) {
+			if(Character.isDigit(str.charAt(index)))
+			num = num+str.charAt(index);
+			else
+			    if(!num.equals("")) {
+				  sum = sum + Integer.parseInt(num);
+				  num = "";
+                }
+			    if(index == str.length()-1 && num!="")
```

Comment:
You can remove length()'s condition, because if you think on this both talks about same thing, right